tests/nlu/featurizers/test_lm_featurizer.py

```diff
@@ -54,37 +82,77 @@ def create_pretrained_transformers_config(
     return config
 
 
-def train_texts(
-    texts: List[Text], model_name: Text, model_weights: Text
+def process_training_text(
+    texts: List[Text],
+    model_name: Text,
+    model_weights: Text,
+    create_language_model_featurizer: Callable[
+        [Dict[Text, Any]], create_language_model_featurizer
+    ],
 ) -> List[Message]:
+    """ Creates a featurizer and process training data """
     config = create_pretrained_transformers_config(model_name, model_weights)
     whitespace_tokenizer = WhitespaceTokenizer()
-    lm_featurizer = LanguageModelFeaturizer(config)
+    lm_featurizer = create_language_model_featurizer(config)
 
     messages = [Message.build(text=text) for text in texts]
     td = TrainingData(messages)
 
     whitespace_tokenizer.train(td)
-    lm_featurizer.train(td)
+    lm_featurizer.process_training_data(td)
     return messages
 
 
-def process_texts(
-    texts: List[Text], model_name: Text, model_weights: Text
+def process_messages(
+    texts: List[Text],
+    model_name: Text,
+    model_weights: Text,
+    create_language_model_featurizer: Callable[
+        [Dict[Text, Any]], create_language_model_featurizer
+    ],
 ) -> List[Message]:
+    """ Creates a featurizer and processes messages """
     config = create_pretrained_transformers_config(model_name, model_weights)
     whitespace_tokenizer = WhitespaceTokenizer()
-    lm_featurizer = LanguageModelFeaturizer(config)
+    lm_featurizer = create_language_model_featurizer(config)
 
     messages = []
     for text in texts:
         message = Message.build(text=text)
         whitespace_tokenizer.process(message)
-        lm_featurizer.process(message)
         messages.append(message)
+    lm_featurizer.process(messages)
     return messages
 
 
+def test_create(
+    default_model_storage: ModelStorage,
+    resource_language_model_featurizer,
+    default_execution_context: ExecutionContext,
+):
+    """ Test new create function """
+    x = LanguageModelFeaturizerGraphComponent.create(
+        config=LanguageModelFeaturizerGraphComponent.get_default_config(),
+        model_storage=default_model_storage,
+        resource=resource_language_model_featurizer,
+        execution_context=default_execution_context,
+    )
+    assert isinstance(x, LanguageModelFeaturizerGraphComponent)
+
+
+def test_create_callable(
+    create_language_model_featurizer: Callable[
+        [Dict[Text, Any]], create_language_model_featurizer
+    ],
+):
+    """ Test create function as a callable
+    This notation will be used for the remaining tests """
+    x = create_language_model_featurizer(
+        LanguageModelFeaturizerGraphComponent.get_default_config()
+    )
+    assert isinstance(x, LanguageModelFeaturizerGraphComponent)
+
+
```

Comment:
I'd challenge if we need this tests as this should be covered by the other tests implicitly

Reply:
Sure. I actually wrote the callable one because I was trying to figure out how to use `Callable` 😂